spec.emu

```diff
@@ -142,6 +126,65 @@ contributors: Google, Ecma International
+      The default calendar for that locale is usually *"persian"*, but an implementation might also support *"gregory"*<del>, *"islamic"*,</del> and *"islamic-civil"*.
+      The Record in the DateTimeFormat [[LocaleData]] internal slot would therefore include a [[fa-IR]] field whose value is a Record like { [[ca]]: « *"persian"*, *"gregory"*, <del>*"islamic"*,</del> *"islamic-civil"* », [[hc]]: « … », [[nu]]: « … » }, along with other locale-named fields having the same value shape but different elements in their Lists.
+    </emu-note>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="ecma402-datetimeformat-objects">
+  <h1>DateTimeFormat Objects</h1>
+
+  <emu-clause id="sec-ecma402-intl-datetimeformat-constructor">
+    <h1>The Intl.DateTimeFormat Constructor</h1>
+
+    <emu-clause id="sec-createdatetimeformat" type="abstract operation" oldids="sec-initializedatetimeformat,sec-todatetimeoptions">
+      <h1>
+        CreateDateTimeFormat (
+          _newTarget_: a constructor,
+          _locales_: an ECMAScript language value,
+          _options_: an ECMAScript language value,
+          _required_: ~date~, ~time~, or ~any~,
+          _defaults_: ~date~, ~time~, or ~all~,
+        ): either a normal completion containing a DateTimeFormat object or a throw completion
+      </h1>
+
+      <dl class="header">
+      </dl>
+
+      <emu-alg>
+        1. Let _dateTimeFormat_ be ? OrdinaryCreateFromConstructor(_newTarget_, *"%Intl.DateTimeFormat.prototype%"*, « [[InitializedDateTimeFormat]], [[Locale]], [[Calendar]], [[NumberingSystem]], [[TimeZone]], [[HourCycle]], [[DateStyle]], [[TimeStyle]], [[DateTimeFormat]], [[BoundFormat]] »).
+        1. Let _hour12_ be *undefined*.
+        1. Let _modifyResolutionOptions_ be a new Abstract Closure with parameters (_options_) that captures _hour12_ and performs the following steps when called:
+          1. Set _hour12_ to _options_.[[hour12]].
+          1. Remove field [[hour12]] from _options_.
+          1. If _hour12_ is not *undefined*, set _options_.[[hc]] to *null*.
+        1. Let _optionsResolution_ be ? ResolveOptions(%Intl.DateTimeFormat%, %Intl.DateTimeFormat%.[[LocaleData]], _locales_, _options_, « ~coerce-options~ », _modifyResolutionOptions_).
+        1. Set _options_ to _optionsResolution_.[[Options]].
+        1. Let _r_ be _optionsResolution_.[[ResolvedLocale]].
+        1. Set _dateTimeFormat_.[[Locale]] to _r_.[[Locale]].
+        1. Let _resolvedCalendar_ be _r_.[[ca]].
+        1. <ins>If _resolvedCalendar_ is *"islamic"* or *"islamic-rgsa"*, then</ins>
+          1. <ins>Let _fallbackCalendar_ be an implementation- and locale-defined calendar type that is one of the values returned from AvailableCalendars.</ins>
+          1. <ins>Set _fallbackCalendar_ to CanonicalizeUValue(*"ca"*, _fallbackCalendar_).</ins>
+          1. <ins>Set _resolvedCalendar_ to _fallbackCalendar_.</ins>
```

Comment:
These are the four added lines in CreateDateTimeFormat. I didn't include anything about a console warning because there isn't really a precedent for ECMA specifying one. But we could choose to add something like "NOTE: Implementations are encouraged to notify the programmer of the fallback using a mechanism such as a log message."